File: Angular/01. Introduction to Angular and TypeScript/demo/05. Boxes/Box.ts
class Box<Type> {
    private _content: Array<Type>;
    constructor() {
        this._content = [];
    }
    public add(element: Type) {
        this._content.push(element)
    };
    public remove() {
        this._content.pop();
    }
    public get count(): number {
        return this._content.length;
    }
}

let box = new Box<Number>();
box.add(1);
box.add(2);
box.add(3);
console.log(box.count);

let box = new Box<String>();
box.add("Pesho");
box.add("Gosho");
console.log(box.count);
box.remove();
console.log(box.count);

File: C# Advanced/08. Generics/Lab/01.Box/Box.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BoxOfT
{
    public class Box<T>
    {
        private List<T> _list;

        public Box()
        {
            _list = new List<T>();
        }

        public void Add(T element)
        {
            _list.Add(element);
        }

        public T Remove()
        {
            T element = _list[_list.Count - 1];
            _list.RemoveAt(_list.Count - 1);
            return element;
        }

        public int Count => _list.Count;
    }
}
